.gitlab/e2e.yml

```diff
@@ -134,6 +134,7 @@ k8s-e2e-otlp-main:
     - aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.ssh_key_rsa --with-decryption --query "Parameter.Value" --out text > $E2E_PRIVATE_KEY_PATH
     # Use S3 backend
     - pulumi login "s3://dd-pulumi-state?region=us-east-1&awssdk=v2&profile=$AWS_PROFILE"
+    - export CI_JOB_NAME="${OVERRIDE_CI_JOB_NAME:-$CI_JOB_NAME}"
```

Comment:
I had to override the `CI_JOB_NAME` variable because the `new_e2e_template` invoke command uses it as the junit tar filename, but when using parallel/matrix jobs the resulting `CI_JOB_NAME` is an invalid filename.
To avoid overriding this variable, what do you think about doing this:
`JUNIT_TAR_FILENAME="${JUNIT_TAR_FILENAME:-$CI_JOB_NAME}"`
and then using the `JUNIT_TAR_FILENAME` variable as the new `--junit-tar` invoke command parameter and setting `JUNIT_TAR_FILENAME` only for parallel jobs?

Reply:
I think there was some requests already about location of Junit files, but I don't remember exactly, cc @pducolin

Reply:
Agree with using another value, I need to check how this is handled in containers tests introduced by @L3n41c.

In [agent-linux-installer repo](https://github.com/DataDog/agent-linux-install-script/blob/main/.gitlab-ci.yml#L261) I do:

```
--junitfile "junit-${CI_JOB_NAME}.xml" 
```

without using the invoke task.

Reply:
How's the name invalid ? Is it an invalid file name ? Could a sanitise processing in the invoke task help ?

Reply:
This resulting file name is:
```
junit-new-e2e-cws-dev: [-c cwsinstrumentation:fullImagePath=669783387624.dkr.ecr.us-east-1.amazonaws.com/cws-instrumentation:26829148-8305f002 --run TestECSFargate].tgz
```
I'm able to reproduce the issue by running this on the agent sandbox:
```
inv -e new-e2e-tests.run --targets ./tests/cws --junit-tar "junit-new-e2e-cws-dev: [-c cwsinstrumentation:fullImagePath=669783387624.dkr.ecr.us-east-1.amazonaws.com/cws-instrumentation:26829148-8305f002 --run TestECSFargate].tgz" --run TestECSFargate
```
I'm not quite sure what's wrong with the file name, but I'll investigate further to see how it could be sanitized.